### ISSUE TITLE
Info regarding number of keypoints for training

### ISSUE THREAD

**Issue by vishalsharbidar** [chenghao-shi/MDGAT-matcher#8]
Hello, 
Thanks for the work and great paper. 

While training, I noticed that the number of keypoints for img1 and img2 is constant. The loss function is also built on this assumption. 
Can you please let me know what may happen if I use a different number of keypoints for img1 and img2? 
Would it have an adverse effect on the training?
For example: If I use 500 keypoints for Img1 and 1000 keypoints for Img2. 

Thank you.

**Comment by chenghao-shi** [comment]
Hi，


We keep the number of keypoints to be constant for the convenience of batch training. It's no need for the number for img1 to be the same as the number for img2. The training is feature-level, more keypoints mean more samples, so I don't think it will have an adverse effect. But after increasing the number of keypoints, the memory usage of the attention increases significantly, which is probably the only thing you need to be aware of.&nbsp;


施成浩
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "chenghao-shi/MDGAT-matcher"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年9月4日(星期天) 晚上8:06
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;[chenghao-shi/MDGAT-matcher] Info regarding number of keypoints for training (Issue #8)





 
Hello,
 Thanks for the work and great paper.
 
While training, I noticed that the number of keypoints for img1 and img2 is constant. The loss function is also built on this assumption.
 Can you please let me know what may happen if I use a different number of keypoints for img1 and img2?
 Would it have an adverse effect on the training?
 For example: If I use 500 keypoints for Img1 and 1000 keypoints for Img2.
 
Thank you.

**Comment by vishalsharbidar** [comment]
Thanks for the input, It was helpful.

**Comment by vishalsharbidar** [comment]
So, when I have a different number of keypoints for image pairs, would it be good to use batch size = 1 for training?

**Comment by chenghao-shi** [comment]
I think using a larger batch size will give better results.

施成浩
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "chenghao-shi/MDGAT-matcher"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年9月5日(星期一) 晚上9:28
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [chenghao-shi/MDGAT-matcher] Info regarding number of keypoints for training (Issue #8)





 
So, when I have different number of keypoints for image pairs, would it be good to use batch size = 1 for training?

**Comment by vishalsharbidar** [comment]
Okay, Thank you
